From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NyCode <nico@nycode.de>
Date: Thu, 1 Jul 2021 20:18:59 +0000
Subject: [PATCH] Configure publishing


diff --git a/build.gradle.kts b/build.gradle.kts
index 7ad3e5153718f6d4ce8293a9790bc3c1158aeb8e..5d7cad5ad8fc8b1955d63eea0ce9d300ea7447eb 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -3,6 +3,7 @@ import java.util.Locale
 plugins {
     `java-library`
     `maven-publish`
+    signing
 }
 
 java {
@@ -84,3 +85,5 @@ tasks.withType<Javadoc>().configureEach {
         // Paper end
     )
 }
+
+apply(from = "../publishing.gradle.kts")
