From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NyCode <nico@nycode.de>
Date: Mon, 26 Jul 2021 00:56:03 +0200
Subject: [PATCH] Add nyper alias


diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
index 708a3411487f59834d049c063af22b43375fa56f..c7551a20fc3b53299111a667e5c2800f4c2237fd 100644
--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
@@ -69,6 +69,7 @@ public class PaperConfig {
         commands = new HashMap<String, Command>();
         commands.put("paper", new PaperCommand("paper"));
         commands.put("mspt", new MSPTCommand("mspt"));
+        commands.put("nyper", new PaperCommand("nyper"));
 
         version = getInt("config-version", 21);
         set("config-version", 21);
diff --git a/src/main/java/de/nycode/nyper/NyperConfig.java b/src/main/java/de/nycode/nyper/NyperConfig.java
new file mode 100644
index 0000000000000000000000000000000000000000..b35e808a75b33af77491f30de0216fc174a9bd5f
--- /dev/null
+++ b/src/main/java/de/nycode/nyper/NyperConfig.java
@@ -0,0 +1,5 @@
+package de.nycode.nyper;
+
+public class NyperConfig {
+
+}
